tests/mocha/touch_test.js

```diff
@@ -0,0 +1,95 @@
+            Blockly.Touch.shouldHandleEvent(mouseEvent1);
+            chai.assert.isTrue(Blockly.Touch.shouldHandleEvent(mouseEvent2));
+        });
+
+        test('does not handle mouseup if not tracking touch', function() {
+            const mouseEvent = new MouseEvent('mouseup');
+            chai.assert.isFalse(Blockly.Touch.shouldHandleEvent(mouseEvent));
+        });
+
+        test('handles mouseup if already tracking a touch', function() {
+            const mousedown = new MouseEvent('mousedown');
+            const mouseup = new MouseEvent('mouseup');
+            // Register the mousedown event first
+            Blockly.Touch.shouldHandleEvent(mousedown);
+            chai.assert.isTrue(Blockly.Touch.shouldHandleEvent(mouseup));
+        });
+
+        test('handles pointerdown if this is a new touch', function() {
+            const pointerdown = new PointerEvent('pointerdown', {pointerId: 1, pointerType: 'touch'});
+            chai.assert.isTrue(Blockly.Touch.shouldHandleEvent(pointerdown));
+        });
+
+        test('does not handle pointerdown if part of a different touch', function() {
+            const pointerdown1 = new PointerEvent('pointerdown', {pointerId: 1, pointerType: 'touch'});
+            const pointerdown2 = new PointerEvent('pointerdown', {pointerId: 2, pointerType: 'touch'});
+            Blockly.Touch.shouldHandleEvent(pointerdown1);
+            chai.assert.isFalse(Blockly.Touch.shouldHandleEvent(pointerdown2));
+        });
+
+        test('does not handle pointerdown after a mousedown', function() {
+            const mouseEvent = new MouseEvent('mousedown');
+            const pointerdown = new PointerEvent('pointerdown', {pointerId: 1, pointerType: 'touch'});
+            Blockly.Touch.shouldHandleEvent(mouseEvent);
+            chai.assert.isFalse(Blockly.Touch.shouldHandleEvent(pointerdown));
+        });
+
+        test('does not handle pointerup if not tracking touch', function() {
+            const pointerup = new PointerEvent('pointerup', {pointerId: 1, pointerType: 'touch'});
+            chai.assert.isFalse(Blockly.Touch.shouldHandleEvent(pointerup));
+        });
+
+        test('handles pointerup if part of existing touch', function() {
+            const pointerdown = new PointerEvent('pointerdown', {pointerId: 1, pointerType: 'touch'});
+            const pointerup = new PointerEvent('pointerdown', {pointerId: 1, pointerType: 'touch'});
+            Blockly.Touch.shouldHandleEvent(pointerdown);
+            chai.assert.isTrue(Blockly.Touch.shouldHandleEvent(pointerup));
+        });
+    });
+
+    suite('getTouchIdentifierFromEvent', function() {
+        test('is mouse for MouseEvents', function() {
+            const mousedown = new MouseEvent('mousedown');
+            chai.assert.equal(Blockly.Touch.getTouchIdentifierFromEvent(mousedown), 'mouse');
+        });
+
+        test('is pointerId for mouse PointerEvents', function() {
```

Comment:
Just curious: why do we want this?

Reply:
So this test basically makes sure the behavior is the same as it was in blockly-v8.0.5. In that code (and now this code), if `e.pointerId` was defined, we'd use that, no matter what. In PointerEvents, it should always be defined, so effectively, we always used that. That means for old old old `MouseEvents` (that are not `PointerEvents`) we'd use the string `'mouse'` but we haven't actually used it for a long time.

I wrote this test because originally I assumed that if the `PointerEvent` had a `pointerType` that was `mouse` then we'd use the string `'mouse'` but that is not correct. We just always use `pointerId` if it's available. So basically I wrote the test as a reminder of the current behavior.